### PR TITLE
Data Management - Feature and Dataset Management Pages

### PR DESCRIPTION
**Pull request by KeremHmd**
- Feature and Dataset Management Pages are implemented with proper backend connections.
- Few other forms have been updated with new design choices.

### CODE REVIEW

File: src/app/modules/data-engineer-page/dataset-management/dataset-management-edit/learning-dataset-creation/learning-dataset-creation-form/learning-dataset-creation-form.component.ts [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+export class LearningDatasetCreationFormComponent extends BaseComponent implements OnInit, OnChanges {
+
+    /** The ID of the dataset */
+    @Input() datasetId: number;
+
+    /** The learning dataset to be edited or created */
+    @Input() learningDataset: LearningDataset;
+
+    /** The transformation associated with the learning dataset */
+    @Input() transformation: DatasetTransformation;
```
Comment (Okanmercan99) [inline src/app/modules/data-engineer-page/dataset-management/dataset-management-edit/learning-dataset-creation/learning-dataset-creation-form/learning-dataset-creation-form.component.ts:25]: Let's pass Transformation id and fetch from API in this module.

File: src/app/modules/data-engineer-page/featureset-management/featureset-management-edit/featureset-details/featureset-details.component.html [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                       placeholder="{{'FeatureSetManagement.URL' | translate}}"
+                       formControlName="featuresetURL"
+                       class="form-control">
+            </div>
+            <!-- Experiment -->
+            <div class="col-2 flex align-items-center mt-4">
+                <label class="text-900 font-medium" for="experiment">{{'FeatureSetManagement.Experiment' | translate}}</label>
+            </div>
+            <div class="col-10 text-900 flex align-items-center p-fluid mt-4">
+                <p-dropdown
```
Comment (Okanmercan99) [inline src/app/modules/data-engineer-page/featureset-management/featureset-management-edit/featureset-details/featureset-details.component.html:45]: Experiment dropdown is not selected although selected featureSet has experiment information.

File: src/app/modules/data-engineer-page/featureset-management/featureset-management-edit/featureset-features/featureset-features-form/featureset-features-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+@Component({
+    selector: 'app-featureset-features-form',
+    templateUrl: './featureset-features-form.component.html',
+    styleUrls: ['./featureset-features-form.component.scss']
+})
+export class FeatureSetFeaturesFormComponent extends BaseComponent implements OnInit, OnChanges {
+    /** The ID of the selected feature set */
+    @Input() selectedFeatureSetId: number;
+    /** The feature to be edited or created */
+    @Input() feature: Feature;
```
Comment (Okanmercan99) [inline src/app/modules/data-engineer-page/featureset-management/featureset-management-edit/featureset-features/featureset-features-form/featureset-features-form.component.ts:19]: Let's pass featureId and fetch feature information from the API.

File: src/app/shared/models/dataset.model.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+export class Dataset {
```
Comment (Okanmercan99) [inline src/app/shared/models/dataset.model.ts:1]: Missing comments.

File: src/app/shared/models/datasetCharacteristic.model.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+export class DatasetCharacteristic {
```
Comment (Okanmercan99) [inline src/app/shared/models/datasetCharacteristic.model.ts:1]: Missing comments.

File: src/app/shared/models/datasetTransformation.model.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+export class DatasetTransformation {
```
Comment (Okanmercan99) [inline src/app/shared/models/datasetTransformation.model.ts:1]: Missing comments.

File: src/app/shared/models/datasetTransformationStep.model.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
+export class DatasetTransformationStep {
```
Comment (Okanmercan99) [inline src/app/shared/models/datasetTransformationStep.model.ts:1]: Missing comments.

File: src/app/shared/models/feature.model.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
+export class Feature {
```
Comment (Okanmercan99) [inline src/app/shared/models/feature.model.ts:1]: Missing comments.

File: src/app/shared/models/featureset.model.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
+export class FeatureSet {
```
Comment (Okanmercan99) [inline src/app/shared/models/featureset.model.ts:1]: missing comments.

File: src/app/shared/models/learningDataset.model.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
+export class LearningDataset {
```
Comment (Okanmercan99) [inline src/app/shared/models/learningDataset.model.ts:1]: missing comments.

**Review by Okanmercan99 — CHANGES_REQUESTED** [review]

File: src/app/modules/dataset-management/dataset-management-edit/learning-dataset-creation/learning-dataset-creation-table/learning-dataset-creation-table.component.ts [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+    /**
+     * Loads the dataset transformations associated with the learning datasets.
+     */
+    loadDatasetTransformations() {
+        this.learningDatasets.forEach(learningDataset => {
+            this.datasetTransformationService.getDatasetTransformationById(learningDataset.dataTransformationId)
+                .pipe(takeUntil(this.destroy$))
+                .subscribe(transformation => {
+                    this.datasetTransformations[learningDataset.dataTransformationId] = transformation;
+                });
```
Comment (Okanmercan99) [inline src/app/modules/dataset-management/dataset-management-edit/learning-dataset-creation/learning-dataset-creation-table/learning-dataset-creation-table.component.ts:97]: Let's add error handling here.

File: src/app/modules/dataset-management/dataset-management-edit/learning-dataset-creation/learning-dataset-creation-table/learning-dataset-creation-table.component.ts [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+
+    /**
+     * Deletes a learning dataset by its ID.
+     * @param learningDataset The learning dataset to be deleted
+     */
+    deleteLearningDataset(learningDataset: LearningDataset) {
+        this.datasetTransformationService.deleteDatasetTransformation(learningDataset.dataTransformationId).pipe(takeUntil(this.destroy$))
+            .subscribe({
+                next: () => {
+                    this.learningDatasets = this.learningDatasets.filter(ld => ld.learningDatasetId !== learningDataset.learningDatasetId);
```
Comment (Okanmercan99) [inline src/app/modules/dataset-management/dataset-management-edit/learning-dataset-creation/learning-dataset-creation-table/learning-dataset-creation-table.component.ts:109]: Let's drop out datasetTransformations related to deleted learningDataset

File: src/app/modules/featureset-management/featureset-management-edit/featureset-features/featureset-features-form/featureset-features-form.component.html [hunk 13]
```diff
@@ -0,0 +1,70 @@
+<p-dialog (onHide)="closeDialog()" [(visible)]="this.display" [modal]="true" [style]="{width: '80rem'}" header="{{ 'FeatureSetManagement.Feature Form' | translate }}">
```
Comment (Okanmercan99) [inline src/app/modules/featureset-management/featureset-management-edit/featureset-features/featureset-features-form/featureset-features-form.component.html:1]: Let's add FeatureSetManagement.Feature Form definition into i18n file.

**Review by Okanmercan99 — CHANGES_REQUESTED** [review]

**Review by Okanmercan99 — CHANGES_REQUESTED** [review]

**Review by Okanmercan99 — APPROVED** [review]

**Review by dogukan10 — APPROVED** [review]

**Pull request merged**